pkg/apis/wksprovider/controller/wksctl/machine_actuator.go

```diff
@@ -479,9 +499,14 @@ func (a *MachineActuator) update(ctx context.Context, cluster *clusterv1.Cluster
 		}
 		return a.kubeadmUpOrDowngrade(machine, node, installer, version, planKey, planJSON, worker)
 	}
-	if err = a.performActualUpdate(installer, machine, node, nodePlan, c); err != nil {
-		return err
-	}
+	//
+	// OK not upgrade or downgrade? so why this?
+	// This is non-logical update becuse it's already done by kubeadm?
```

Comment:
This was intended to support updates such as turning a working into a master, etc. that doesn't fall under the "upgrade" heading.

Reply:
reverted